compound_disease/compound_treats_disease/word_vector_experiment/generate_word_vectors.py

```diff
@@ -0,0 +1,244 @@
+
+# coding: utf-8
+
+# # Generate Word Vectors For Compound Treats Disease Sentences
+
+# This notebook is designed to generate word vectors for gene interacts gene (CtD) sentences. Using facebooks's fasttext, we trained word vectors using all sentences that contain a disease and gene mention. The model was trained using the following specifications:
+# 
+# | Parameter | Value |
+# | --- | --- |
+# | Size | 300 |
+# | alpha | 0.005 | 
+# | window | 2 |
+# | epochs | 50 |
+# | seed | 100 | 
+
+# # Set Up Environment
+
+# In[2]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+get_ipython().run_line_magic('matplotlib', 'inline')
+
+from collections import defaultdict
+import os
+import pickle
+import sys
+
+sys.path.append(os.path.abspath('../../../modules'))
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+from tqdm import tqdm_notebook
+
+from gensim.models import FastText
+from gensim.models import KeyedVectors
+
+from utils.notebook_utils.dataframe_helper import load_candidate_dataframes, generate_embedded_df
+
+
+# In[3]:
+
+
+#Set up the environment
+username = "danich1"
+password = "snorkel"
+dbname = "pubmeddb"
+
+#Path subject to change for different os
+database_str = "postgresql+psycopg2://{}:{}@/{}?host=/var/run/postgresql".format(username, password, dbname)
+os.environ['SNORKELDB'] = database_str
+
+from snorkel import SnorkelSession
+session = SnorkelSession()
+
+
+# In[4]:
+
+
+from snorkel.learning.pytorch.rnn.rnn_base import mark_sentence
+from snorkel.learning.pytorch.rnn.utils import candidate_to_tokens
+from snorkel.models import Candidate, candidate_subclass
+
+
+# In[5]:
+
+
+CompoundDisease = candidate_subclass('CompoundDisease', ['Compound', 'Disease'])
+
+
+# # Compound Treats Disease
+
+# This section loads the dataframe that contains all compound treats disease candidate sentences and their respective dataset assignments.
+
+# In[9]:
+
+
+cutoff = 300
+total_candidates_df = (
+    pd
+    .read_table("../dataset_statistics/results/all_ctd_map.tsv.xz")
+    .query("sen_length < 300")
+)
+total_candidates_df.head(2)
+
+
+# # Train Word Vectors
+
+# This section trains the word vectors using the specifications described above.
```

Comment:
Not sure how necessary this is because perhaps future people looking at this PR will be more familiar with NLP, but I am a bit lost about what is the input and output of your training and embedding steps.  I am also not sure how they fit into your bigger picture - how are the embeddings used downstream?

Reply:
**tl;dr** - for a computer to utilize sentences in a machine learning sentence we need to "embed" them into a vector space. Input: all individual words found in sentences -> output: vector space with dimensions [total vocab size, word vector size (typically somewhere between 50-300)]

The end goal is to train a model to classify sentences; however, a computer can only work with numbers and not words. To fix this problem one has to use a coding scheme to essentially "embed" sentences into a vector space a computer can use. 

One way is to do a one hot encoding where you have large dimensional vectors (the total vocab size that one is working with. i.e. 300k+) and place a 1 at the index that represents a given word. The downside as you can imagine is this method doesn't perform quite as well in some cases. 

Despite one hot embedding, another way to "embed" sentences is to use [word2vec](https://towardsdatascience.com/introduction-to-word-embedding-and-word2vec-652d0c2060fa) which is a fancy way of saying train a neural network to learn the context of sentences and then generate embeddings for each word.  <- The project works with this idea.

Ultimately, I use these word embeddings to train a convolutional neural network that will classify the sentences I am working with.